modules/apps/commerce/commerce-product-type-virtual-service/src/main/java/com/liferay/commerce/product/type/virtual/service/impl/CPDefinitionVirtualSettingServiceImpl.java

```diff
@@ -201,14 +211,13 @@ private void _checkCommerceCatalog(long cpDefinitionId, String actionId)
 			getPermissionChecker(), commerceCatalog, actionId);
 	}
 
-	private static volatile ModelResourcePermission<CommerceCatalog>
-		_commerceCatalogModelResourcePermission =
-			ModelResourcePermissionFactory.getInstance(
-				CPDefinitionVirtualSettingServiceImpl.class,
-				"_commerceCatalogModelResourcePermission",
-				CommerceCatalog.class);
+	@Reference(
+		target = "(model.class.name=com.liferay.commerce.product.model.CommerceCatalog)"
```

Comment:
@yuhai My only concern is why this class uses the resource permission of `CommerceCatalog`, but for the conversion it's good to go.